src/libprojectM/MilkdropPresetFactory/CustomWave.hpp

```diff
@@ -18,109 +18,99 @@
  * See 'LICENSE.txt' included within this release
  *
  */
-/**
- * $Id$
- *
- * Encapsulation of a custom wave
- *
- * $Log$
- */
-
-#ifndef _CUSTOM_WAVE_H
-#define _CUSTOM_WAVE_H
+#pragma once
 
 #define CUSTOM_WAVE_DEBUG 0
 
-class CustomWave;
 class Expr;
 class PerPointEqn;
 class Preset;
 
-#include <vector>
+#include "Renderer/Waveform.hpp"
 
 #include "Common.hpp"
-#include "Param.hpp"
-#include "PerFrameEqn.hpp"
-#include "Renderer/Waveform.hpp"
+
+#include "MilkdropPresetFactory/Param.hpp"
+#include "MilkdropPresetFactory/PerFrameEqn.hpp"
 
 #include <map>
+#include <vector>
 
 class CustomWave : public Waveform
 {
 public:
+     CustomWave() = delete;
```

Comment:
Deletes the default constructor, otherwise the compiler might generate a default one. It's a C++11 feature:

https://en.cppreference.com/w/cpp/language/function#Deleted_functions